server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -276,8 +294,27 @@ public void parse(ParseContext context) throws IOException {
                 context.doc().add(field);
             }
         } catch (Exception e) {
-            throw new MapperParsingException("failed to parse field [{}] of type [{}] in document with id '{}'", e, fieldType().name(),
-                    fieldType().typeName(), context.sourceToParse().id());
+            String valuePreview = "";
+            try {
+
+                XContentParser parser = context.parser();
+                int tokenDepth = 10;
```

Comment:
Is it necessary to define `tokenDepth` here or can it be defined instead inside `readComplexJsonElement`?